src/Microsoft.Extensions.Caching.Memory/MemoryCache.cs

```diff
@@ -87,6 +88,12 @@ public ICacheEntry CreateEntry(object key)
 
         private void SetEntry(CacheEntry entry)
         {
+            if (_disposed)
+            {
+                // No-op instead of throwing since this is called during CacheEntry.Dispose
+                return;
+            }
+
```

Comment:
Is this related to the `ConcurrentDictionary` change, or was it a bug in the previous implementation as well?